### DISCUSSION TITLE
lldb-vscode and "custom" launch type

### DISCUSSION THREAD

**Discussion by mellery451** [mfussenegger/nvim-dap#730] [Q&A]
I'm trying to get lldb-vscode working for a remote debug scenario in which I manually launch the remote debugger and inferior process in a single command. I can't figure out the corresponding nvim-dap config to then attach to that lldb-server (I can do it fine manually with lldb though). [this document](https://github.com/vadimcn/vscode-lldb/blob/master/MANUAL.md#remote-debugging) seems to indicate that there should be a "custom" launch type that enables detailed control of the lldb commands via target/processCreateCommands, but I just can't get this to work in nvim-dap.

Here's my config so far:
```
        {
            name = '(lldb) remote',
            type = 'lldb',
            request = 'custom',
            initCommands = {
                "platform select remote-linux",
                "platform connect connect://localhost:2159",
                "settings set target.source-map /proc/self/cwd /home/mellery/myproj",
            },
            processCreateCommands = {
                "continue",
            },
        },
```
and this results in a message in the debugger window:

```
terminate called without an active exception
```
..and no connection is made to the lldb-server.

If I switch to request = 'launch', then it does seem like there is a connection made to the server, but debugging doesn't actually work and it gets disconnected soon after intial connect. The debugger window shows an error like:

```
[-] warning: get_frame() failed. Is the target binary started?
```

I should clarify, the way I'm launching the lldb-server (in a local docker container) is basically by running something like this in the container:

```
lldb-server platform --server --listen *:2159' -- <program> <args>
```

Any suggestions about what else to try?

**Comment by LiadOz** [comment]
I commented here https://github.com/mfussenegger/nvim-dap/discussions/600#discussioncomment-4069958 my config for remote debugging using cppdbg.
I start lldb-server like this 
```
lldb-server platform --server --listen 0.0.0.0:1337
```
start `a.out` and I change the pid in the config for the correct one.
